core/chains/evm/client/errors.go

```diff
@@ -76,14 +76,16 @@ var parity = ClientErrors{
 
 // Geth
 // See: https://github.com/ethereum/go-ethereum/blob/b9df7ecdc3d3685180ceb29665bab59e9f614da5/core/tx_pool.go#L516
-var gethFatal = regexp.MustCompile(`(: |^)(exceeds block gas limit|invalid sender|negative value|oversized data|gas uint64 overflow|intrinsic gas too low|nonce too high)$`)
+// Updated error messages to not have `$` in the regex to support extra information in the error messages
+//    see: https://github.com/ethereum/go-ethereum/blob/a9ef135e2dd53682d106c6a2aede9187026cc1de/core/state_transition.go#L223-L224
+var gethFatal = regexp.MustCompile(`(: |^)(exceeds block gas limit|invalid sender|negative value|oversized data|gas uint64 overflow|intrinsic gas too low|nonce too high)`)
 var geth = ClientErrors{
-	NonceTooLow:                       regexp.MustCompile(`(: |^)nonce too low$`),
-	ReplacementTransactionUnderpriced: regexp.MustCompile(`(: |^)replacement transaction underpriced$`),
+	NonceTooLow:                       regexp.MustCompile(`(: |^)nonce too low`),
+	ReplacementTransactionUnderpriced: regexp.MustCompile(`(: |^)replacement transaction underpriced`),
 	TransactionAlreadyInMempool:       regexp.MustCompile(`(: |^)(?i)(known transaction|already known)`),
-	TerminallyUnderpriced:             regexp.MustCompile(`(: |^)transaction underpriced$`),
-	InsufficientEth:                   regexp.MustCompile(`(: |^)(insufficient funds for transfer|insufficient funds for gas \* price \+ value|insufficient balance for transfer)$`),
-	TxFeeExceedsCap:                   regexp.MustCompile(`(: |^)tx fee \([0-9\.]+ [a-zA-Z]+\) exceeds the configured cap \([0-9\.]+ [a-zA-Z]+\)$`),
+	TerminallyUnderpriced:             regexp.MustCompile(`(: |^)transaction underpriced`),
+	InsufficientEth:                   regexp.MustCompile(`(: |^)(insufficient funds for transfer|insufficient funds for gas \* price \+ value|insufficient balance for transfer)`),
+	TxFeeExceedsCap:                   regexp.MustCompile(`(: |^)tx fee \([0-9\.]+ [a-zA-Z]+\) exceeds the configured cap \([0-9\.]+ [a-zA-Z]+\)`),
```

Comment:
This requires great care since some of this stuff can actually erroneously match other errors. e.g. if you loosen up “transaction underpriced” too much it can match “replacement transaction underpriced” which is a different error.

Would recommend simply adding an optional terminal string instead for each case e.g.

```
NonceTooLow: regexp.MustCompile(`(: |^)nonce too low(address 0x[a-fA-F0-0]{32}, tx: \d+ state: \d+)?$`)```

Reply:
@samsondav we would prefer to fix this error in a more forward compatible way if possible. Encoding another overly-constrained match only handles this specific case and leaves us open to the same thing happening again.

Reply:
And we are only loosening to allow more characters `after` the error, not before. Meaning it wouldn't match `replacement transaction underpriced` if we ever loosen up the `transaction underpriced` check.

Reply:
It looks like we have this option of mirroring the prefix match on semicolon as an in-between :shrug: 
``(: |^)nonce too low(:|$)`